src/main/java/org/folio/circulation/domain/LoanCheckInService.java

```diff
@@ -16,4 +16,13 @@
       : loan.checkIn(request.getCheckInDate(), systemDateTime,
         request.getServicePointId()));
   }
+
+  public boolean isInHouseUse(Item item, RequestQueue requestQueue,
+    CheckInByBarcodeRequest checkInByBarcodeRequest) {
+
+    return item.isAvailable()
+      && (requestQueue == null || requestQueue.size() == 0)
```

Comment:
Thank you for extracting this to a place that can better facilitate testing.

Please can tests be added to cover both conditions on this line (Sonar reports that only one is covered).